hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ReadReplicas.java

```diff
@@ -246,4 +204,91 @@ private File createDirectory(String volumeName, String bucketName,
   }
+
+  private void findCandidateKeys(OzoneClient ozoneClient, OzoneAddress address) throws IOException {
+    ObjectStore objectStore = ozoneClient.getObjectStore();
+    String volumeName = address.getVolumeName();
+    String bucketName = address.getBucketName();
+    String keyName = address.getKeyName();
+    if (!keyName.isEmpty()) {
+      processKey(ozoneClient, volumeName, bucketName, keyName);
+    } else if (!bucketName.isEmpty()) {
+      OzoneVolume volume = objectStore.getVolume(volumeName);
+      OzoneBucket bucket = volume.getBucket(bucketName);
+      checkBucket(bucket, ozoneClient);
+    } else if (!volumeName.isEmpty()) {
+      OzoneVolume volume = objectStore.getVolume(volumeName);
+      checkVolume(volume, ozoneClient);
+    } else {
+      for (Iterator<? extends OzoneVolume> it = objectStore.listVolumes(null); it.hasNext();) {
+        checkVolume(it.next(), ozoneClient);
+      }
+    }
+  }
+
+  private void checkVolume(OzoneVolume volume, OzoneClient ozoneClient) throws IOException {
+    for (Iterator<? extends OzoneBucket> it = volume.listBuckets(null); it.hasNext();) {
+      OzoneBucket bucket = it.next();
+      checkBucket(bucket, ozoneClient);
+    }
+  }
+
+  private void checkBucket(OzoneBucket bucket, OzoneClient ozoneClient) throws IOException {
+    String volumeName = bucket.getVolumeName();
+    String bucketName = bucket.getName();
+    for (Iterator<? extends OzoneKey> it = bucket.listKeys(null); it.hasNext();) {
+      OzoneKey key = it.next();
+//    TODO: Remove this check once HDDS-12094 is fixed
+      if (key.getName().endsWith("/")) {
+        continue;
+      }
+      processKey(ozoneClient, volumeName, bucketName, key.getName());
+    }
+  }
+
+  private void processKey(OzoneClient client, String volumeName, String bucketName, String keyName) throws IOException {
+    System.out.println("Processing key : " + volumeName + "/" + bucketName + "/" + keyName);
+    boolean isChecksumVerifyEnabled
+        = getConf().getBoolean("ozone.client.verify.checksum", true);
+    OzoneConfiguration configuration = new OzoneConfiguration(getConf());
+    configuration.setBoolean("ozone.client.verify.checksum",
+        !isChecksumVerifyEnabled);
+    RpcClient newClient = new RpcClient(configuration, null);
+
+    try {
+      ClientProtocol noChecksumClient;
+      ClientProtocol checksumClient;
+      if (isChecksumVerifyEnabled) {
+        checksumClient = client.getObjectStore().getClientProxy();
+        noChecksumClient = newClient;
+      } else {
+        checksumClient = newClient;
+        noChecksumClient = client.getObjectStore().getClientProxy();
+      }
+
+      File dir = createDirectory(volumeName, bucketName, keyName);
+      OzoneKeyDetails keyInfoDetails = checksumClient.getKeyDetails(volumeName, bucketName, keyName);
+      Map<OmKeyLocationInfo, Map<DatanodeDetails, OzoneInputStream>> replicas =
+          checksumClient.getKeysEveryReplicas(volumeName, bucketName, keyName);
```

Comment:
Not totally related, but RpcClient.getKeysEveryReplicas() doesn't seems to create input stream that refreshes container cache upon failure. Could that lead problems in such a corner case?

https://github.com/apache/ozone/blob/master/hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java#L1609